### PR TITLE
1440934: Ensure rhsmcertd performs both types of checks

### PR DESCRIPTION
**Pull request by cnsnyder**
This fixes a regression introduced by the RFE in #1577.

Please note this PR might conflict with #1579 (similar changes were made to ensure the random splay values are actually used). This was done in this PR to ensure the reviewer could successfully test the PR as is. Whenever this, or the other PR is merged, the other can be rebased.

Note to future self: Mind your pointers.

### CODE REVIEW

**Comment by kahowell** [comment]
@cnsnyder, can you please rebase?

**Comment by cnsnyder** [comment]
On it!

File: src/daemons/rhsmcertd.c [hunk 1]
```diff
@@ -256,24 +256,22 @@ cert_check (gboolean heal)
...
-    if (cert_data->heal) {
-        g_timeout_add (cert_data->heal_interval_seconds * 1000,
-            (GSourceFunc) cert_check, (gpointer) cert_data->heal);
-    }
-    else {
-        g_timeout_add (cert_data->cert_interval_seconds * 1000,
-            (GSourceFunc) cert_check, (gpointer) cert_data->heal);
-    };
+    g_timeout_add (cert_data->heal_interval_seconds * 1000,
+        (GSourceFunc) cert_check, (gpointer) heal);
```
Comment (kahowell) [inline src/daemons/rhsmcertd.c]: I noticed we're doing some weird stuff here with casting `gpointer` to `bool` and vice-versa. I added a card to the backlog for this. It's fairly hacky. But this PR doesn't make us any worse than the status quo (been this way for a long time), so I think we should address it later.

**Comment by kahowell** [comment]
Looks good to me.

**Pull request merged**